components/user-store/org.wso2.carbon.identity.user.store.configuration.ui/src/main/java/org/wso2/carbon/identity/user/store/configuration/ui/client/UserStoreConfigAdminServiceClient.java

```diff
@@ -108,27 +119,65 @@ public void deleteUserStoresSet(String[] userStores) throws Exception {
         stub.deleteUserStoresSet(userStores);
     }
 
+    /**
+     * Deletes a given list of user stores from repository
+     * @param userStores : an instance of an array of {@link UserStoreDTO}
+     * @throws RemoteException RemoteException
+     * @throws RemoteException UserStoreConfigAdminServiceIdentityUserStoreMgtException UserStoreConfigAdminServiceException
+     */
+    public void deleteUserStoresSet(UserStoreDTO[] userStores) throws RemoteException,
+            UserStoreConfigAdminServiceIdentityUserStoreMgtException {
+        stub.deleteUserStoresSetFromRepository(userStores);
+    }
+
     /**
      * Deletes a given user store
      *
      * @param userStore : domain name of the user store to deleted
      * @throws Exception
      */
+    @Deprecated
     public void deleteUserStore(String userStore) throws Exception {
         stub.deleteUserStore(userStore);
     }
 
+    /**
+     * Delete user store from any repository.
+     * @param userStoreDTO an instance of {@link UserStoreDTO}
+     * @throws RemoteException RemoteException
+     * @throws UserStoreConfigAdminServiceIdentityUserStoreMgtException UserStoreConfigAdminServiceException
+     */
+    public void deleteUserStoreFromRepository(UserStoreDTO userStoreDTO) throws RemoteException,
+            UserStoreConfigAdminServiceIdentityUserStoreMgtException {
+        stub.deleteUserStoreFromRepository(userStoreDTO);
+    }
+
     /**
      * Toggle user store state (enable/disable)
      *
      * @param domain     : domain name of the user store to enable/dissable
      * @param isDisabled : set true to disable user store
-     * @throws Exception
+     * @throws RemoteException
      */
+    @Deprecated
     public void changeUserStoreState(String domain, String isDisabled) throws Exception {
         stub.changeUserStoreState(domain, Boolean.parseBoolean(isDisabled));
     }
 
+    /**
+     * Toggle user store state (enable/disable)
+     *
+     * @param domain     : domain name of the user store to enable/dissable
+     * @param isDisabled : set true to disable user store
+     * @param repositoryClasses : the repository where the userstore properties are persisted.
+     * @throws RemoteException if an error occured while changing the userstore state.
+     */
+    public void enableDisableUserStore(String domain, String isDisabled, String repositoryClasses) throws RemoteException,
```

Comment:
Can't we use the same exiting name changeUserStoreState(String domain, String isDisabled, String repositoryClasses)